### PR TITLE
Feature/53 make sidebar collapsible

### PR DESCRIPTION
**Pull request by AdrianBorg**
Few changes from what was there before; this PR aimed at making the sidebar collapsible. On a big screen it would be on the left and have an icon to dismiss it, and another icon to bring it back. This was done using material ui drawers

In order for this to work more easily, the structure of the `home` component was changed. Also adding a simple header that we could make better in another ticket

also added the package clsx which makes it easier to add conditional classNames, and the react-window-hook package in order to conveniently get the size of the window within a react component.

I'm not the biggest fan of how I needed to do a bunch of css setup for the Material UI transitions (and couldnt have them completely css). Also not entirely sure I've implemented drawers completely correctly, because it was pretty fiddly to get them to work with the rest of the page (and resizing the main content area when its collapsed).

Switch from completely grid layout in the `home` component because its not possible in most browsers to have an animated transition where the number of grid columns changes (when the sidebar goes from open to collapsed) so switched to flexbox.

Couple gifs of what goes on:
Full screen drawer:

Smaller screen drawer:

### CODE REVIEW

**Comment by AdrianBorg** [comment]
Just realised that I've got to fix the tests. I'll have a look and do that later

File: frontend/src/constants/sidebarOptions.ts [hunk 1]
```diff
@@ -4,6 +4,13 @@ const sidebarOptions = [
   { name: 'Employees', endpoint: 'employees' },
 ];
 
+export const SIDEBAR_STATES = {
+  CLOSED: 'closed',
+  CLOSING: 'closing',
+  OPEN: 'open',
+  OPENING: 'opening',
+};
+
```
Comment (tjanowsk) [inline frontend/src/constants/sidebarOptions.ts:13]: Is this used anywhere?

File: frontend/src/components/header/header.module.scss [hunk 2]
```diff
@@ -0,0 +1,21 @@
+@import '../../constants/styles.scss';
+
+.headerContent {
+  grid-area: header;
+  background-color: $colour-lightBlue;
+  display: grid;
+  grid-template-areas: 'left middle right';
+  grid-template-columns: 2.5rem 8fr 80px;
```
[the remark this answers is not in the document]
Comment (AdrianBorg) [reply frontend/src/components/header/header.module.scss:8]: there's no difference, think I had other `fr`s but changed them to different units and forgot to switch this

File: frontend/src/components/sidebar/index.tsx [hunk 3]
```diff
@@ -1,35 +1,107 @@
...
   ));
 
-const Sidebar = () => (
-  <div className={sidebarStyles.container}>
-    {getButtons(sidebarOptions)}
-    <LoginOutButton />
-  </div>
-);
+const Sidebar = () => {
+  const [open, setOpen] = useState(true);
```
[the remark this answers is not in the document]
Comment (AdrianBorg) [reply frontend/src/components/sidebar/index.tsx:62]: thats right, I'll need to switch that over

File: frontend/src/constants/sidebarOptions.ts (hunk 1 above)
Comment (AdrianBorg) [reply frontend/src/constants/sidebarOptions.ts:13]: ah used it in a previous iteration and forgot to remove

**Pull request merged**